controllers/common.go

```diff
@@ -80,36 +89,66 @@ func PopulateResticSecret(name string, namespace string, label string) (*corev1.
 
 func BuildResticSecret(givensecret *corev1.Secret, secret *corev1.Secret, resticrepo string) error {
 
-	// assign new restic secret values
-	for key, val := range givensecret.Data {
-		switch {
-		case key == AWSAccessKey:
-			AWSAccessValue = val
-		case key == AWSSecretKey:
-			AWSSecretValue = val
-		case key == AWSDefaultRegion:
-			AWSDefaultRegionValue = val
-		case key == ResticPassword:
-			ResticPasswordValue = val
+	provider := givensecret.Labels[OADPBSLProviderName]
+
+	switch provider {
+	case AWSProvider:
+		// assign new restic secret values
+		for key, val := range givensecret.Data {
+			switch {
+			case key == AWSAccessKey:
+				AWSAccessValue = val
+			case key == AWSSecretKey:
+				AWSSecretValue = val
+			case key == AWSDefaultRegion:
+				AWSDefaultRegionValue = val
+			case key == ResticPassword:
+				ResticPasswordValue = val
+			}
 		}
-	}
 
-	// build new Restic secret
-	resticSecretData := &corev1.Secret{
-		Data: map[string][]byte{
-			AWSAccessKey:     AWSAccessValue,
-			AWSSecretKey:     AWSSecretValue,
-			AWSDefaultRegion: AWSDefaultRegionValue,
-			ResticPassword:   ResticPasswordValue,
-			ResticRepository: []byte(resticrepo),
-		},
+		// build new Restic secret
+		resticSecretData := &corev1.Secret{
+			Data: map[string][]byte{
+				AWSAccessKey:     AWSAccessValue,
+				AWSSecretKey:     AWSSecretValue,
+				AWSDefaultRegion: AWSDefaultRegionValue,
+				ResticPassword:   ResticPasswordValue,
+				ResticRepository: []byte(resticrepo),
+			},
+		}
+		secret.Data = resticSecretData.Data
+		return nil
+
+	case AzureProvider:
+		// assign new restic secret values
+		for key, val := range givensecret.Data {
+			switch {
+			case key == AzureAccountName:
+				AzureAccountNameValue = val
+			case key == AzureAccountKey:
+				AzureAccountKeyValue = val
+			case key == ResticPassword:
+				ResticPasswordValue = val
+			}
+		}
+
+		// build new Restic secret
+		resticSecretData := &corev1.Secret{
+			Data: map[string][]byte{
+				AzureAccountName: AzureAccountNameValue,
+				AzureAccountKey:  AzureAccountKeyValue,
+				ResticPassword:   ResticPasswordValue,
+				ResticRepository: []byte(resticrepo),
+			},
+		}
+		secret.Data = resticSecretData.Data
+		return nil
```

Comment:
- Add a case for empty provider labe
- Add a default case for unsupported providers

Reply:
@kaovilai we do not support empty provider or any other provider other than aws/azure/gcp, its already validated in bsl.go https://github.com/openshift/oadp-operator/blob/master/controllers/bsl.go#L61